src/diffusers/loaders/lora_base.py

```diff
@@ -898,3 +898,17 @@ def lora_scale(self) -> float:
         # property function that returns the lora scale which can be set at run time by the pipeline.
         # if _lora_scale has not been set, return 1
         return self._lora_scale if hasattr(self, "_lora_scale") else 1.0
+
+    def enable_lora_hotswap(self, **kwargs) -> None:
+        """Enables the possibility to hotswap LoRA adapters.
+
+        Calling this method is only required when hotswapping adapters and if the model is compiled or if the ranks of
+        the loaded adapters differ.
```

Comment:
Wonder if it makes sense to rename it to `enable_lora_hotswap_for_compilation()`? This way, there's no ambiguity amongst users. Otherwise, I feel like users may get the impression that they may need to call this method even when they are not using compilation. 

WDYT?